Puesdo Hooks/XComGame/XGTacticalGameCore/GetOverheatIncrement.txt

```diff
@@ -0,0 +1,81 @@
+//This script was generated by HexToPseudoCode decompiler for use with PatchUPK/PatcherGUI tool
+UPK_FILE = XComGame.upk
+OBJECT = XGTacticalGameCore.GetOverheatIncrement : AUTO
+[REPLACEMENT_CODE]
+/*(0x0000/0x0000)*/ 0B 
+/*(0x0001/0x0001)*/ 07 [@label_0x0012] 9A 00 <.iWeapon> 25 16 
+/*(0x0010/0x000C)*/ 	04 25 
+/*(0x0012/0x000E)*/ [#label_0x0012]
+/*(0x0012/0x000E)*/ 07 [@label_0x002C] 1B <WeaponHasProperty> 00 <.iWeapon> 2C 0A 16 
+/*(0x002A/0x0022)*/ 	04 25 
+/*(0x002C/0x0024)*/ [#label_0x002C]
+/*(0x002C/0x0024)*/ 07 [@label_0x0046] 1B <WeaponHasProperty> 00 <.iWeapon> 2C 12 16 
+/*(0x0044/0x0038)*/ 	04 25 
+/*(0x0046/0x003A)*/ [#label_0x0046]
+/*(0x0046/0x003A)*/ 07 [@label_0x0060] 1B <WeaponHasProperty> 00 <.iWeapon> 2C 0D 16 
+/*(0x005E/0x004E)*/ 	04 25 
+/*(0x0060/0x0050)*/ [#label_0x0060]
+/*(0x0060/0x0050)*/ 0F 00 <.iAmount> 35 <XGTacticalGameCoreNativeBase.TWeapon.iSuppression> <XGTacticalGameCoreNativeBase.TWeapon> 00 00 1B <GetTWeapon> 00 <.iWeapon> 16 
+/*(0x0090/0x0070)*/ 0B 
+/*(0x0091/0x0071)*/ 0B 
+/*(0x0092/0x0072)*/ 07 [@label_0x00D5] 97 9C 1A 2C 7B 35 <XGTacticalGameCoreNativeBase.TCharacter.aUpgrades> <XGTacticalGameCoreNativeBase.TCharacter> 00 00 48 <.kCharacter> 2C 02 16 25 16 
+/*(0x00BB/0x008F)*/ 	0F 00 <.iAmount> 91 00 <.iAmount> 2C 64 16 
+/*(0x00D2/0x009E)*/ 	06 [@label_0x00EC] 
+/*(0x00D5/0x00A1)*/ [#label_0x00D5]
+/*(0x00D5/0x00A1)*/ 0F 00 <.iAmount> FD 00 <.iAmount> 2C 64 16 
+/*(0x00EC/0x00B0)*/ [#label_0x00EC]
+/*(0x00EC/0x00B0)*/ 07 [@label_0x01A6] 81 1B <WeaponHasProperty> 00 <.iWeapon> 2C 02 16 16 
+/*(0x0106/0x00C6)*/ 	07 [@label_0x0137] 97 1A 2C 3A 35 <XGTacticalGameCoreNativeBase.TCharacter.aUpgrades> <XGTacticalGameCoreNativeBase.TCharacter> 00 00 48 <.kCharacter> 25 16 
+/*(0x012B/0x00DF)*/ 		A1 00 <.iAmount> 26 16 
+/*(0x0137/0x00E7)*/ 	[#label_0x0137]
+/*(0x0137/0x00E7)*/ 	07 [@label_0x016E] 1C <XGTacticalGameCoreNativeBase.TInventoryHasItemType> 35 <XGTacticalGameCoreNativeBase.TCharacter.kInventory> <XGTacticalGameCoreNativeBase.TCharacter> 00 00 48 <.kCharacter> 2C 50 16 
+/*(0x0162/0x0102)*/ 		A1 00 <.iAmount> 26 16 
+/*(0x016E/0x010A)*/ 	[#label_0x016E]
+/*(0x016E/0x010A)*/ 	07 [@label_0x01A6] 1C <XGTacticalGameCoreNativeBase.TInventoryHasItemType> 35 <XGTacticalGameCoreNativeBase.TCharacter.kInventory> <XGTacticalGameCoreNativeBase.TCharacter> 00 00 48 <.kCharacter> 2C 7B 16 
+/*(0x0199/0x0125)*/ 		A1 00 <.iAmount> 2C 02 16 
+/*(0x01A6/0x012E)*/ [#label_0x01A6]
+
+/*
+            XComGameInfo(WorldInfo.Game).Mods[0].Mods("RoulettePlus.RoulettePlusMod").Object(self, true);
```

Comment:
sending `self` here is `XGTacticalGameCore` which is not useful for this hook.
consider sending `kUnit`.